agpypeline/geometries.py

```diff
@@ -128,15 +133,16 @@ def geometry_to_geojson(geom: ogr.Geometry, alt_coord_type: str = None, alt_coor
                 'code': ref_sys.GetAttrValue("AUTHORITY", 1)
             }
         }
-
     return json.dumps(geom_json)
 
 
-def polygon_from_ring(ring: ogr.Geometry, epsg: int = None) -> Optional[ogr.Geometry]:
+def polygon_from_ring(ring: ogr.Geometry, epsg: int = None, filename: str = None) -> Optional[ogr.Geometry]:
```

Comment:
I thought we had talked about getting rid of the `filename` parameter? This should return None when there's a problem and let the calling function do what it thinks is correct